lightningd/options.c

```diff
@@ -191,6 +191,38 @@ static char *opt_add_addr(const char *arg, struct lightningd *ld)
 	return opt_add_addr_withtype(arg, ld, ADDR_LISTEN_AND_ANNOUNCE, true);
 }
 
+static char *opt_alt_subdaemon(const char *arg, struct lightningd *ld)
+{
+	// example arg: "hsmd:rmthsmd"
+	char *argbuf = tal_strdup(ld, arg);
+	char *colon = strchr(argbuf, ':');
+	if (!colon)
+	{
+		tal_free(argbuf);
+		return tal_fmt(NULL, "argument must contain \':\'");
+	}
+	*colon = '\0';
+
+	if (!is_subdaemon(argbuf))
+	{
+		char *ret = tal_fmt(NULL, "%s is not a subdaemon", argbuf);
+		tal_free(argbuf);
+		return ret;
+	}
+	char *subdaemon = tal_strdup(ld, argbuf);
```

Comment:
It's preferable to declare it at the top of the function ?

Reply:
Not sure about this.  It needs to be after argbuf is modified by replacing the ':' w/ '/0'

Do you mean declared at the top and then initialized later?

Reply:
Generally it is declared at the start of a scope (not necessarily start of a function).

Reply:
fixed